backend/src/controllers/device.controller.ts

```diff
@@ -126,6 +126,44 @@ export const updateTask = (req: Request, res: Response) => {
     DeviceModel.findOne(conditions, callback);
 };
 
+export const updateTaskInProgress = (req: Request, res: Response) => {
+    const conditions = {
+        deviceId: req.params.deviceId,
+    };
+
+    const callback = (err: any, device: DeviceInterface) => {
+        if (err) {
+            res.status(500).send(err.message);
+        } else if (!device) {
+            res.status(404).send('Device not found');
+        } else {
+            // finds the task in the device's tasks
+            const taskIndex = device.tasks.findIndex((task: TaskInterface) => task._id == req.params.taskId);
+            if (taskIndex === -1) {
+                res.status(404).send('Task not found');
+            } else {
+                if(typeof req.body.inProgress !== "boolean"){
+                    res.status(400).send('Value of inProgress is not a boolean')
+                } else {
```

Comment:
Tu zamiast zagnieżdżonych ifów można użyć `else if`, to będzie przejrzyściej

Reply:
Done!